### PR TITLE
Add because/becauseArgs to `Contain(IEnumerable<T>)`

### LINKED ISSUE #806: Contain(IEnumerable<T>, params T[]) lacks because parameter
The method

```cs
public AndConstraint<TAssertions> Contain(IEnumerable<T> expectedItemsList, params T[] additionalExpectedItems);
```
in the class `SelfReferencingCollectionAssertions<T, TAssertions>` lacks a `string because` parameter. Probably because `additionalExpectedItems` is a `params`.

Suggestions:

1. Refactor to:

```cs
public AndConstraint<TAssertions> Contain(IEnumerable<T> expectedItemsList, IEnumerable<T>additionalExpectedItems, string because = "", params object[] becauseArgs);
```

Pro's:
- Same signature as other methods

Cons:
- Breaking change

2. Add:

```cs
public AndConstraint<TAssertions> Contain(IEnumerable<T> expectedItemsList, string because ="", params T[] additionalExpectedItems);
```

Pro's:
- Non-breaking change

Cons:
- No `params object[] becauseArgs`
- Still does not match other method signatures

### PR DESCRIPTION
**Pull request by jnyrup**
fb7c6970 added `Contain(IEnumerable<T> originalItems, params T[] additionalItems)`
which left out the ability to provide `because` and `becauseArgs`.

This fixes #806

### CODE REVIEW

File: Src/FluentAssertions/Collections/StringCollectionAssertions.cs [hunk 1]
```diff
@@ -261,42 +261,6 @@ public AndConstraint<TAssertions> Contain(IEnumerable<string> expected)
...
-        /// </summary>
-        /// <param name="expected">An <see cref="IEnumerable{T}" /> with the expected elements.</param>
-        /// <param name="because">
-        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
-        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
-        /// </param>
-        /// <param name="becauseArgs">
-        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
-        /// </param>
-        public AndConstraint<TAssertions> Contain(IEnumerable<string> expected, string because = null,
```
Comment (jnyrup) [inline Src/FluentAssertions/Collections/StringCollectionAssertions.cs:276]: These two are removed as they now come for free through `Contain(IEnumerable<T>)` from the base class.

File: Src/FluentAssertions/Collections/GenericDictionaryAssertions.cs [hunk 2]
```diff
@@ -652,7 +652,7 @@ public AndConstraint<TAssertions> Contain(params KeyValuePair<TKey, TValue>[] ex
         /// <param name="becauseArgs">
         /// Zero or more objects to format using the placeholders in <paramref name="because" />.
         /// </param>
-        public AndConstraint<TAssertions> Contain(IEnumerable<KeyValuePair<TKey, TValue>> expected,
+        public new AndConstraint<TAssertions> Contain(IEnumerable<KeyValuePair<TKey, TValue>> expected,
```
Comment (dennisdoomen) [inline Src/FluentAssertions/Collections/GenericDictionaryAssertions.cs:655]: Why did you need this?

File: Src/FluentAssertions/Collections/SelfReferencingCollectionAssertions.cs [hunk 3]
```diff
@@ -439,15 +439,17 @@ public AndConstraint<TAssertions> EndWith(IEnumerable<T> expectation, string bec
...
-            params T[] additionalExpectedItems)
+        /// <param name="expected">An <see cref="IEnumerable{T}"/> of expectation items.</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because"/>.
+        /// </param>
+        public AndConstraint<TAssertions> Contain(IEnumerable<T> expected, string because = "", params object[] becauseArgs)
```
Comment (dennisdoomen) [inline Src/FluentAssertions/Collections/SelfReferencingCollectionAssertions.cs:450]: I don't get this change. You seem to be replacing an overload that takes an additional set of items to include in the expected collection by the overload that takes a reason.

**Comment by jnyrup** [comment]
I must have misread #806.
I thought there was an actual problem, but it seems to more of a wondering of why there wasn't a generic overload `Contain(IEnumerable<T>, string, params object[])` available.

As mentioned in the issue, it is currently handled by the non-generic `Contain(IEnumerable, string, params object[])`.
(At least for generic expectations also implementing `IEnumerable`)
